File: Python_Workspace/369Game.py

def clap(th):
    for j in range(0,th) : print("clap ",end='')
    print()
    
def gamestart(num):
    for i in range(1,num+1):
        index = str(i)
        cnt=0
        th=0
        for j in range(0,len(index)):
            if int(index[j]) == 3 or int(index[j]) == 6 or int(index[j]) == 9:
                cnt = 1
                th += 1
        if(cnt != 0):
            clap(th)
        else :
            print(i)

def main():
    num = input("Please input the last number"+'\n')
    print("--------start--------")
    gamestart(int(num))
    
if __name__ == "__main__":
    main()

File: C_Workspace/369game.cpp
#include <stdio.h>
#include <stdlib.h>

void clap(int th){
	int i;
	for(i=0;i<th;i++) printf("clap! ");
	printf("\n");
}
void gamestart(int num){
	int i,j,count,cnt,th,m;
	int index[10];
	for(i=1;i<=num;i++){
		count = 0;
		m = i;
		while(m>0){
			index[count++] = m % 10;
			m = m / 10;
		}
		cnt = 0;
		th = 0;
		for(j=0;j<count;j++){
			if(index[j] == 3 || index[j] == 6 || index[j] == 9){
				cnt = 1;
				th++;
			}
		}
		if(cnt != 0) clap(th);
		else printf("%d\n",i);
	}
	
}
int main(){
	int num;
	
	printf("Please input the last number\n");
	scanf("%d",&num);
	printf("--------start--------\n");
	gamestart(num);
	return 0;	
}